backtracking/all_combinations.py

```diff
@@ -56,6 +68,28 @@ def create_all_state(
     current_list: list[int],
     total_list: list[list[int]],
 ) -> None:
+    """
+    Helper function to recursively build all combinations.
+
+    >>> create_all_state(1, 4, 2, [], result := [])
```

Comment:
I personally don't like how this helper function modifies an input argument rather than returning a new list, but that's not an issue with this PR and should be left for future work.